src/game.cpp

```diff
@@ -10908,16 +10908,38 @@ void game::eat(int pos)
         }
         return;
     }
-    if (pos == INT_MIN) {
-        pos = inv_type(_("Consume item:"), IC_COMESTIBLE);
+
+    if( pos != INT_MIN ) {
+        u.consume(pos);
+        return;
     }
 
-    if (pos == INT_MIN) {
+    auto filter = [&]( const item &it ) {
+        return u.inv.has_category( it, IC_COMESTIBLE, u );
```

Comment:
Or just simply: `return it.is_food(&u) || it.is_food_container(&u))` - same as in `inventory::has_category`
Actually, the line 10912 above seems to be the only place that `IC_COMESTIBLE` is used.  It might be good to remove that enumeration value completely and only use the inlined lambdas.

At the very least: `has_category` is static and does not need `u.inv.` in front of it.